src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -422,6 +424,33 @@ private static void ThrowInvalidDocumentId(string id)
             }
             else
             {
+                // check for attachments with duplicate name
+                if ((flags & DocumentFlags.Resolved) == DocumentFlags.Resolved)
+                {
+                    if (metadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray currentDocumentAttachments))
+                    {
+                        // actualAttachments can contain attachments of current document which were directly put, we want to check if there are attachment with duplicate name
+                        var attachmentsChanged = false;
+                        foreach (BlittableJsonReaderObject bjro in currentDocumentAttachments)
+                        {
+                            var currentAttachment = JsonDeserializationClient.AttachmentName(bjro);
+
+                            if (actualAttachments.Items.Count(o =>
+                                o is DynamicJsonValue attachment && attachment[nameof(AttachmentName.Name)] is LazyStringValue lzv && lzv == currentAttachment.Name) > 1)
```

Comment:
The types here are _very_ exact. Can it be other things? 
This looks very fragile.

Reply:
this is returned from the attachments storage:
https://github.com/ravendb/ravendb/blob/ff622ad5b817a4245e8011e0a5779599236f1511/src/Raven.Server/Documents/AttachmentsStorage.cs#L146
and the values are from table

Reply:
That is fragile. If we'll change this in the future, there is no indication that this is related.

Reply:
done